### PR TITLE
erhise/dev

### LINKED ISSUE #5: Prevent ts file from exiting
Find some clever way to make the ts file keep running after the developer written code is finished, so that it can be run again on changes to the file. Ideally this should be handled outside of the file the developer is currently editing and be applied wrapping the code being tested by the developer.

### PR DESCRIPTION
**Pull request by erhise**
* Added default tsconfig.json with `tsc --init`
  * need to discuss or agree on what config should include
* Added nodemon config, [see reference commit for description](https://github.com/erhise/offline-blitz/pull/6/commits/8e4a9ef45b355bb12b7f072200c60ece0e4a5658)
* Added package-lock.json to .gitignore since it will generate "unecessary" git diffs (open for discussion)
* Update scripts in package.json
  * `npm run sandbox`: will run app.ts
  * `npm run copy:default`: will replace app.ts with app.template.ts
* Added some suggestion on extending template
  * maybe remove default `unused import` rule

image

Ideas for future
* Add some default header that will run `console.clear()` on each restart? This will able to clear potential previous console logs.
* `npm run sandbox` should as a suggestion be able to take `npm run sandbox -- my-file-name` which should run my-file-name.ts in watch mode.
  * I briefly looked up some implementation suggestions => Was not sure about best way to achieve this, but some kind of script file to look at arguments and return correct execution.
  * Add directory named `personal` to `src/personal`? Where`npm run sandbox -- my-file-name` would point to `src/personal/my-file-name.ts`

Closes #5

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,7 +4,9 @@
   "description": "1. clone repo 2. install dependencies 3. run script 4. clean app.ts",
   "main": "index.js",
   "scripts": {
-      "blitz": "cp template/app.template.ts src/app.ts && ./node_modules/ts-node/dist/bin.js src/app.ts"
+    "blitz": "cp templates/app.template.ts src/app.ts && ./node_modules/ts-node/dist/bin.js src/app.ts",
```
Comment (erhise) [inline package.json:7]: maybe we can remove this but call sandbox blitz instead?

File: templates/rxjs.template.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+// === rxjs.template.ts =============
+console.clear(); // clear console for each save
+import { of } from 'rxjs';
+import { map, filter } from 'rxjs/operators';
```
Comment (erhise) [inline templates/rxjs.template.ts:4]: I was thinking that a small set of common operators could be initially imported

File: package.json (hunk 1 above)
Comment (siguse) [reply package.json:7]: Yes, we can. It nodemon was actually supposed to be before the ts-node part. That's also the reason that I created #5. Let's just forget that and use this sandbox script instead :D

**Review by siguse — APPROVED** [review]

**Comment by siguse** [comment]
I approve, you may want to remove the old "blitz" script before merging. :)

**Pull request merged**